tooling/minimize/src/enums.rs

```diff
@@ -0,0 +1,121 @@
+use crate::*;
+mod rs {
+    pub use crate::rs::*;
+    pub use crate::rustc_target::abi::{Variants, FieldsShape, Primitive, TagEncoding, VariantIdx};
+}
+
+pub fn translate_enum<'tcx>(
+    ty: rs::Ty<'tcx>,
+    adt_def: rs::AdtDef<'tcx>,
+    sref: rs::GenericArgsRef<'tcx>,
+    tcx: rs::TyCtxt<'tcx>,
+) -> Type {
+    let a = rs::ParamEnv::empty().and(ty);
+    let layout = tcx.layout_of(a).unwrap().layout;
+    let size = translate_size(layout.size());
+    let align = translate_align(layout.align().abi);
+
+    let Type::Int(discriminant_ty) = translate_ty(ty.discriminant_ty(tcx), tcx) else {
+        panic!("Discriminant type is not integer!")
+    };
+
+    let (variants, discriminator) = match layout.variants() {
+        rs::Variants::Single { index } => {
+            let fields = translate_fields(layout.fields(), adt_def.variant(*index), sref, tcx);
+            ([(Int::ZERO, Variant { ty: Type::Tuple { fields, size, align }, tagger: Map::new() })].into_iter().collect(), Discriminator::Known(0.into()))
```

Comment:
```suggestion
            let variants = [(Int::ZERO, Variant { ty: Type::Tuple { fields, size, align }, tagger: Map::new() })];
            let discriminator = Discriminator::Known(0.into());
            (variants.into_iter().collect::<Map<_, _>>(), discriminator)
```
That makes this a lot more readable